typescript types broken on zod schema plugin on delegated tables in zenstack v2.5.1

zenstack v2.5.1 introduces new error. 

for example for this schema:
```prisma
generator client {
    provider = "prisma-client-js"
    binaryTargets = ["native", "rhel-openssl-3.0.x"]
}

datasource db {
    provider = "postgresql"
    url      = env("DATABASE_URL")
}

plugin zod {
    provider = '@core/zod'
}



model Animal {
    id        String   @id @default(uuid())
    animalType          String            @default("")
    @@delegate(animalType)
}

model Dog extends Animal {
    name String
}
```

you get the following error:
```
 ~/development/projects/tmp/zenstack-sample-issue ············································································································································································································································ 6s  19:12:41 ─╮
❯ zenstack  generate                                                                                                                                                                                                                                                                                           ─╯
⌛️ ZenStack CLI v2.5.1, running plugins
✔ Generating Prisma schema
✔ Generating PrismaClient enhancer
✔ Generating Zod schemas
Error compiling generated code:
node_modules/.pnpm/@zenstackhq+runtime@2.5.1_@prisma+client@5.15.1_prisma@5.15.1_/node_modules/.zenstack/zod/models/Animal.schema.ts:41:23 - error TS2322: Type 'boolean' is not assignable to type 'never'.

41             id: true, animalType: true
                         ~~~~~~~~~~
node_modules/.pnpm/@zenstackhq+runtime@2.5.1_@prisma+client@5.15.1_prisma@5.15.1_/node_modules/.zenstack/zod/models/Animal.schema.ts:49:23 - error TS2322: Type 'boolean' is not assignable to type 'never'.

49             id: true, animalType: true
                         ~~~~~~~~~~
node_modules/.pnpm/@zenstackhq+runtime@2.5.1_@prisma+client@5.15.1_prisma@5.15.1_/node_modules/.zenstack/zod/models/Dog.schema.ts:43:23 - error TS2322: Type 'boolean' is not assignable to type 'never'.

43             id: true, animalType: true
                         ~~~~~~~~~~
node_modules/.pnpm/@zenstackhq+runtime@2.5.1_@prisma+client@5.15.1_prisma@5.15.1_/node_modules/.zenstack/zod/models/Dog.schema.ts:51:23 - error TS2322: Type 'boolean' is not assignable to type 'never'.

51             id: true, animalType: true
                         ~~~~~~~~~~

: Error compiling generated code

```

removing zod plugin or @@delegate field would relax this issue. currently we decided not to upgrade because we are using zod schemas in our app.

**Environment (please complete the following information):**

-   ZenStack version: 2.5.1
-   Prisma version: 5.15.1
-   Database type: Postgresql

--- a/packages/schema/src/plugins/zod/generator.ts
+++ b/packages/schema/src/plugins/zod/generator.ts
@@ -370,10 +370,10 @@ export function ${refineFuncName}<T, D extends z.ZodTypeDef>(schema: z.ZodType<T
             }
 
             // delegate discriminator fields are to be excluded from mutation schemas
-            const delegateFields = model.fields.filter((field) => isDiscriminatorField(field));
+            const delegateDiscriminatorFields = model.fields.filter((field) => isDiscriminatorField(field));
             const omitDiscriminators =
-                delegateFields.length > 0
-                    ? `.omit({ ${delegateFields.map((f) => `${f.name}: true`).join(', ')} })`
+                delegateDiscriminatorFields.length > 0
+                    ? `.omit({ ${delegateDiscriminatorFields.map((f) => `${f.name}: true`).join(', ')} })`
                     : '';
 
             ////////////////////////////////////////////////
@@ -485,9 +485,11 @@ export const ${upperCaseFirst(model.name)}PrismaUpdateSchema = ${prismaUpdateSch
 
             // mark fields with default as optional
             if (fieldsWithDefault.length > 0) {
+                // delegate discriminator fields are omitted from the base schema, so we need
+                // to take care not to make them partial otherwise the schema won't compile
                 createSchema = this.makePartial(
                     createSchema,
-                    fieldsWithDefault.map((f) => f.name)
+                    fieldsWithDefault.filter((f) => !delegateDiscriminatorFields.includes(f)).map((f) => f.name)
                 );
             }
 

--- a/tests/regression/tests/issue-1693.test.ts
+++ b/tests/regression/tests/issue-1693.test.ts
@@ -1,0 +1,20 @@
+import { loadSchema } from '@zenstackhq/testtools';
+
+describe('issue 1693', () => {
+    it('regression', async () => {
+        await loadSchema(
+            `
+            model Animal {
+                id String @id @default(uuid())
+                animalType String @default("")
+                @@delegate(animalType)
+            }
+
+            model Dog extends Animal {
+                name String
+            }
+            `,
+            { fullZod: true }
+        );
+    });
+});

